content/docs/command-reference/exp/branch.md

```diff
@@ -15,17 +15,19 @@ positional arguments:
 
 ## Description
 
-Makes a
-[Git branch](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
-off the last commit (`HEAD`) based on the given `experiment`, using the `branch`
-name provided. This makes the given experiment
-[persistent](/doc/user-guide/experiment-management#persistent-experiments) in
-the repo.
-
-In most cases this is similar to using `dvc exp apply` (applies the `experiment`
-results to the workspace) followed by Git branching and committing, except that
-`dvc exp branch` **does not** switch into the created `branch`. Equivalent to
-this:
+Makes a named Git
+[`branch`](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
+containing the target `experiment` (making it
+[persistent](/doc/user-guide/experiment-management#persistent-experiments)). The
+branch is based on the commit that was active (`HEAD`) at the time that the
+target experiment was run.
+
+This command is useful, for example, in cases where `dvc exp apply` fails due to
+significant diversion between the project version at the time of the experiment
+run and the current <abbr>workspace</abbr>.
```

Comment:
I would say there are a few other uses:
1. Wanting to preserve multiple experiments by creating a branch from each.
2. Wanting to preserve experiments without needing to apply them to the workspace (if you're in the middle of experimenting with other things).
3. Wanting to preserve checkpoint commits in your Git history (instead of squashing them like `exp apply` would do).
It's not necessary to be exhaustive, but I actually think that `exp branch` primarily for these reasons rather than as a backup to `exp apply`.

Reply:
Yeah, very good summary! I put this info in (with some addendums) in 2510027e, but still keeping the `exp apply` contrasting 😬